mailpoet/lib/WooCommerce/CouponPreProcessor.php

```diff
@@ -91,4 +91,11 @@ private function generateRandomCode(): string {
   private function shouldGenerateCoupon(array $block): bool {
     return empty($block['couponId']);
   }
+  
+  /**
+   * For some renders/send outs the coupon id shouldn't be persisted along the coupon block
+   */
+  private function shouldPersist(NewsletterEntity $newsletter): bool {
+    return $newsletter->getType() !== NewsletterEntity::TYPE_AUTOMATIC;
```

Comment:
Aha, so that means that for automatic emails, we will generate a coupon for every sending? Correct?
I think that TYPE_AUTOMATION emails and maybe also TYPE_WELCOME potentially also  TYPE_RE_ENGAGEMENT and TYPE_CONFIRMATION_EMAIL_CUSTOMIZER have the same behavior that they can be sent after some event.

Reply:
Aha. If it is a placeholder, maybe we could always return true and add a comment that in the future, it will return true for dynamic coupons. Or maybe just add a comment. When I saw the code it was suspicious. Why only automatic emails?